File: schoolBusTracking-flutter-react-springboot-firebase-main/hopeon_admin/src/service/driverService.js
import http from "./httpService";

const endPoint = "/driver";

export async function findAllDrivers() {
  return await http.get(endPoint + "/findAll");
}

export async function findAllUnAssigned() {
  return await http.get(endPoint + "/findAllUnAssigned");
}

export async function saveDriver(driver) {
  return await http.post(endPoint, driver);
}

export async function updateDriver(driver) {
  return await http.put(endPoint, driver);
}

export async function assignVehicle(vehicleId, driverId){
    return await http.put(
      endPoint + "/assignVehicle?id=" + driverId + "&vehicleId="+ vehicleId
    );
}

File: hopeon_backend/src/main/java/com/backend/hopeOn/service/DriverService.java
package com.backend.hopeOn.service;

import com.backend.hopeOn.domain.Driver;
import com.backend.hopeOn.generic.HOResponse;
import jakarta.mail.MessagingException;

import java.util.List;

public interface DriverService {
    HOResponse<List<Driver>> findAll();
    HOResponse<List<Driver>> findAllUnAssigned();
    HOResponse<Driver> findById(Long id);
    HOResponse<List<Driver>> search(Driver driverSearch);
    HOResponse<Driver> assignVehicle( Long id, Long vehicleId);
    HOResponse<Driver> save(Driver driver) throws MessagingException;
    HOResponse<Driver> update(Driver driver);
    HOResponse<String> delete(Long id);
}
